Objects/genericaliasobject.c

```diff
@@ -180,14 +180,16 @@ tuple_extend(PyObject **dst, Py_ssize_t dstindex,
 PyObject *
 _Py_make_parameters(PyObject *args)
 {
-    Py_ssize_t nargs = PyTuple_GET_SIZE(args);
+    assert (PyTuple_Check(args) || PyList_Check(args));
+    Py_ssize_t nargs = PySequence_Length(args);
     Py_ssize_t len = nargs;
     PyObject *parameters = PyTuple_New(len);
     if (parameters == NULL)
         return NULL;
     Py_ssize_t iparam = 0;
+    const int is_args_tuple = PyTuple_Check(args);
```

Comment:
```suggestion
    const bool is_args_tuple = PyTuple_Check(args);
```
(may need to `#include <stdbool.h>` if it's not already there)